app/javascript/components/import-image-form.jsx

```diff
@@ -0,0 +1,100 @@
+
+      API.options('/api/providers').then(({data: {supported_providers}}) => {
+          var provider_classes = supported_providers;
+
+          API.get('/api/providers?expand=resources&attributes=id,name,type').then(({resources}) => {
+              resources.forEach((provider) => {
+                  if(provider['id'] === ManageIQ.record.recordId) return;
+
+                  var result = provider_classes.find(provider_class => provider_class['type'] === provider['type']);
+
+                  if (typeof result !== typeof undefined && result['kind'] === kind)
+                      options.push({value: provider['id'], label: provider['name']});
+              })
+
+              resolve(options);
+          })
+      })
+  })
+}
+
+const fetchImages = (provider) => {
+    return new Promise((resolve, reject) => {
+        API.get('/api/cloud_templates?expand=resources&attributes=id,name&filter[]=ems_id=' + provider).then(({resources}) => {
+            let options = resources.map(({id, name}) => ({value: id, label: name}));
+            options.unshift(BLANK);
+            resolve(options);
+        })
+    })
+}
+
+const fetchDiskTypes = () => {
+    return new Promise((resolve, reject) => {
+        API.get('/api/cloud_volume_types?expand=resources&attributes=id,name').then(({resources}) => {
+            let options = resources.map(({id, name}) => ({value: id, label: name}));
+            options.unshift(BLANK);
+            resolve(options);
+        })
+    })
+}
+
+const fetchBuckets = (provider) => {
+    return new Promise((resolve, reject) => {
+        API.get('/api/cloud_object_store_containers?expand=resources&attributes=name,ems_id&filter[]=ems_id=' + provider).then(({resources}) => {
+            let options = resources.map(({id, name}) => ({value: id, label: name}));
+            options.unshift(BLANK);
+            resolve(options);
+        })
+    })
+}
+
+const ImportImageForm = ({ dispatch }) => {
+  const [provider, setProvider] = useState('-1');
+  const [storage,  setStorage]  = useState('-1');
+  const [image,    setImage]    = useState('-1');
+  const [bucket,   setBucket]   = useState('-1');
+  const [diskType, setDiskType] = useState('-1');
+  const [keepOva,  setKeepOva]  = useState(false);
+
+  const providers = fetchProviders('cloud');
+  const storages  = fetchProviders('storage');
+  const diskTypes = fetchDiskTypes();
+  const images    = useMemo(() => fetchImages(provider), [provider]);
+  const buckets   = useMemo(() => fetchBuckets(storage), [storage]);
+
+  const initialize = (formOptions) => {
+    dispatch({ type: "FormButtons.init",        payload: { newRecord: true, pristine: true } });
+    dispatch({ type: "FormButtons.customLabel", payload: __('Import') });
+    dispatch({ type: 'FormButtons.callbacks',   payload: { addClicked: () => formOptions.submit() }});
+  };
+
+  const submitValues = () => {
+      if ([provider, image, storage, bucket].some(item => parseInt(item) === -1))
```

Comment:
I really couldn't understand why we need to store values in these variables, these will be available in  values.src_provider_id,values.src_image_id....., You can refer https://github.com/ManageIQ/manageiq-ui-classic/tree/master/app/javascript/components/subnet-form or any other form for reference.

Reply:
I have changed the code and it submits "values" directly + two other json elements that do not come from the UI form / user choice.